pkg/iac/terraform/state/backend/http_reader.go

```diff
@@ -31,6 +33,16 @@ func NewHTTPReader(rawURL string, opts *Options) (*HTTPBackend, error) {
 		return nil, err
 	}
 
+	buf := new(bytes.Buffer)
+	_, err = buf.ReadFrom(res.Body)
+	if err != nil {
```

Comment:
Which kind of errors are covered there ?

Reply:
I think the only possible path is when `client.Do(req)` returns a invalid response body object. Should we ignore this error?

Reply:
I think it may introduce a weakness if we do not handle this case but we should find a way to test this behavior.

Let's close this discussion there and we'll see in your other refacto PR if we can handle that.